docs/src/main/asciidoc/hibernate-search-guide.adoc

```diff
@@ -0,0 +1,627 @@
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'The Hotel New Hampshire', 1);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'The Cider House Rules', 1);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'A Prayer for Owen Meany', 1);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'Last Night in Twisted River', 1);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'In One Person', 1);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'Avenue of Mysteries', 1);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'The New York Trilogy', 2);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'Mr. Vertigo', 2);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'The Brooklyn Follies', 2);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'Invisible', 2);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), 'Sunset Park', 2);
+INSERT INTO book(id, title, author_id) VALUES (nextval('hibernate_sequence'), '4 3 2 1', 2);
+----
+
+== Preparing the infrastructure
+
+We need a PostgreSQL instance and an Elasticsearch cluster.
+
+Let's use Docker to start one of each:
+
+[source, shell]
+----
+docker run -it --rm=true --name elasticsearch_quarkus_test -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.0.1
+----
+
+[source, shell]
+----
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name postgresql_quarkus_test -e POSTGRES_USER=quarkus_test -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test -p 5432:5432 postgres:11.3
+----
+
+== Time to play with your application
+
+You can now interact with your REST service:
+
+ * start Quarkus with `./mvnw compile quarkus:dev`
+ * open a browser to `http://localhost:8080/`
+ * search for authors or book titles (we initialized some data for you)
+ * create new authors and books and search for them too
+
+As you can see, all your updates are automatically synchronized to the Elasticsearch cluster.
+
+== Building a native executable
+
+You can build a native executable with the usual command `./mvnw package -Pnative`.
+
+Running it is as simple as executing `./target/hibernate-search-elasticsearch-1.0-SNAPSHOT-runner`.
+
+You can then point your browser to `http://localhost:8080/` and use your application.
+
+[NOTE]
+====
+The startup is a bit slower than usual: it is mostly due to us dropping and recreating the database schema and the Elasticsearch mapping every time at startup.
+We also inject some data and execute the mass indexer.
+
+In a real life application, it is obviously something you won't do at startup.
```

Comment:
I want to do better here once we have configuration profiles. I would start an empty application which does not drop the database schema/Elasticsearch mapping every time.